Tagging & cleanup of individual OVS settings

slyon opened the pull request:
## Description
This PR implements the "Common tags for all device types" section of the spec.
Whenever any OVS setting is applied inside the ovsdb, this code adds another setting for the corresponding interface inside the `external-ids` table, e.g.:
* `ovs-vsctl set-ssl client-key client-cert ca-cert`
* `ovs-vsctl set open_vswitch . external-ids:netplan/global/set-ssl=client-key,client-cert,ca-cert`
or:
* `ovs-vsctl set Bridge ovs0 other-config:some-key=42`
* `ovs-vsctl set Bridge ovs0 external-ids:netplan/other-config/some-key=42`
or:
* `ovs-vsctl set Port ovs1 lacp=off`
* `ovs-vsctl set Port ovs1 external-ids:netplan/lacp=off`

The netplan settings tag (`netplan/<column>[/<key>]=<value>`) marks the settings (`column[:key]=value`) inside ovsdb state, so that it can be cleaned up via `netplan apply --only-ovs-cleanup`. The cleanup will only happen if the setting and value are exactly the same as set/tagged by netplan. If the setting was touched manually (or by another process) netplan cleanup will not touch it. This way we can make sure to always start with a clean state before executing the OVS setup commands and submitting the new config/settings to ovsdb.

Additionally, this PR fixes the creation and tagging of OVS patch_ports and the setup of global OpenFlow protocols.

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Closes an open bug in Launchpad.

sil2100 reviewed (approved):
Ok, this was yet another big and tiring review, but thank you for preparing this changeset! I had a few minor inline comments but other than that it's all good to go (but none of those are blocking issues). 

In overall I'm a bit annoyed that we even have to do such things like the settings tagging, because this introduces extra complexity. You did a really good job in dealing with it, but it's still so sad that we have to go this extra mile to get things cleaned up and tracked properly. Not much we can do about it though.

Let's get this merged soon!

slyon commented on src/openvswitch.c:
```diff
@@ -105,6 +105,28 @@ netplan_type_is_physical(const NetplanDefType type)
+    g_assert(value);
+    g_autofree char *clean_value = g_strdup(value);
+    /* Replace " " -> "," if value contains spaces */
+    if (strstr(value, " ")) {
```
Indeed!

slyon commented on src/openvswitch.c:
```diff
@@ -105,6 +105,28 @@ netplan_type_is_physical(const NetplanDefType type)
+    /* Replace " " -> "," if value contains spaces */
+    if (strstr(value, " ")) {
+        char **split = g_strsplit(value, " ", -1);
+        clean_value = g_strjoinv(",", split);
```
Thanks! I wasn't really aware of this `g_autofree` characteristic, but it makes sense, as glib doesn't know what we do in between definition of the variable and `return`. Fixed.

slyon commented on src/openvswitch.c:
```diff
@@ -280,6 +321,8 @@ write_ovs_conf(const NetplanNetDefinition* def, const char* rootdir)
                 /* Set LACP mode, default to "off" */
                 append_systemd_cmd(cmds, OPENVSWITCH_OVS_VSCTL " set Port %s lacp=%s",
                                    def->id, def->ovs_settings.lacp? def->ovs_settings.lacp : "off");
+                write_ovs_tag_setting(def->id, type, "lacp", NULL,
```
Yes, I actually thought about that as well some time ago, but I must have lost track of it. Fixed.

slyon commented:
Thanks a log Lukasz! I know this got pretty big, unfortunately... All of this OVS handling has so many special cases, which need to be handled individually. Yeah.. I'm not sure this is the best way to go long term, as almost every 2nd OVS setting is special in any way and needs individual tagging and cleanup. But I cannot think of any other solution at the moment.

So I fixed all you inline comments and re-run all unit- and integration-tests. This brought up one little issue (in the non-ovs integration tests), with `netplan apply` failing if OVS was not installed. Fixed that as well.

I will now merge this PR.
